File: Game_Segments/soln1.rb
def GameSegments(nums)
    sum = 0
    for i in nums do
        if i == 0
            sum-=1
        else
            sum+=1
        end
    end
    return 0 if sum < 0
        
    curr = 0
    for j in (0...nums.size) do
        if nums[j] == 0
            curr-=1
        else
            curr+=1
        end

        if curr > sum - curr
            return j+1
        end
    end
end

x = [1, 1, 0, 1]
print(GameSegments(x))

File: Game_Segments/soln1.c
int GameSegments(int* nums){

    int sum = 0;
    int numsSize = sizeof(nums)/sizeof(int);

    for(int i = 0; i < numsSize; i++)
    {
        nums[i] == 0 ? (sum-=1) : (sum+=1);
    }

    if(sum < 0) return 0;

    int curr = 0;

    for(int j = 0; j < numsSize; j++)
    {
        nums[j] == 0 ? (curr-=1) : (curr+=1);
        if(curr > (sum - curr)) return j+1;
    }
    return;
}
int main(){

    int coins[] = {1, 1, 0, 1};

    printf("%d\n", GameSegments(x));

    return 0;
}